SplunkRumWorkspace/SplunkRum/SplunkRumTests/UtilsTests.swift

```diff
@@ -206,4 +206,55 @@ class UtilsTests: XCTestCase {
 
     }
 
+    /**Test Initialization of SessionBasedSampler**/
+    func testSessionBasedSamplingInitialization() throws {
+        // Forces RUM to reinitialze for testing
+        SplunkRum.initialized = false
+        _ = SplunkRum.initialize(beaconUrl: "http://127.0.0.1:8989/",
+                                 rumAuth: "FAKE_RUM_AUTH",
+                                 options: SplunkRumOptions(allowInsecureBeacon: true,
+                                                           debug: true,
+                                                           globalAttributes: [:],
+                                                           environment: nil,
+                                                           ignoreURLs: nil,
+                                                           sessionBaseSamplingRatio: 0.2)
+        )
+        XCTAssertTrue(SessionBasedSampler.probability >= 0.0 && SessionBasedSampler.probability <= 1.0)
+        XCTAssertEqual(SessionBasedSampler.probability, 0.2)
+    }
+
+    /**Test Sending All Spans**/
+    func testSessionBasedSampling100Pct() throws {
+        // Forces RUM to reinitialze for testing
+        SplunkRum.initialized = false
+        _ = SplunkRum.initialize(beaconUrl: "http://127.0.0.1:8989/",
+                                 rumAuth: "FAKE_RUM_AUTH",
+                                 options: SplunkRumOptions(allowInsecureBeacon: true,
+                                                           debug: true,
+                                                           globalAttributes: [:],
+                                                           environment: nil,
+                                                           ignoreURLs: nil,
+                                                           sessionBaseSamplingRatio: 1.0)
+        )
+        let shouldSample = SessionBasedSampler.sessionShouldSample()
+        XCTAssertTrue(shouldSample)
+    }
+
+    /**Tests Sending 0 Spans**/
+    func testSessionBasedSampling0Pct() throws {
+        // Forces RUM to reinitialze for testing
+        SplunkRum.initialized = false
+        _ = SplunkRum.initialize(beaconUrl: "http://127.0.0.1:8989/",
+                                 rumAuth: "FAKE_RUM_AUTH",
+                                 options: SplunkRumOptions(allowInsecureBeacon: true,
+                                                           debug: true,
+                                                           globalAttributes: [:],
+                                                           environment: nil,
+                                                           ignoreURLs: nil,
+                                                           sessionBaseSamplingRatio: 0.0)
+        )
+        let shouldSample = SessionBasedSampler.sessionShouldSample()
+        XCTAssertFalse(shouldSample)
+    }
+
```

Comment:
There should be a test that, if we ask 100 times with the ratio set to 0.5, that we get between 40 and 60 samples.

Reply:
Test Added.